src/components/TodoAppHeader/TodoAppHeader.tsx

```diff
@@ -0,0 +1,59 @@
+import {
+  FC,
+  memo,
+  useEffect,
+  useRef,
+} from 'react';
+import cn from 'classnames';
+import { filterTodosByCompletion } from '../../utils/todoUtils';
+import { TodoAppHeaderProps } from './TodoAppHeaderProps';
+
+export const TodoAppHeader: FC<TodoAppHeaderProps> = memo(({
+  todos,
+  handleUpdate,
+  handleSubmit,
+  todoTitle,
+  handleQueryChange,
+  isInputDisabled,
+}) => {
+  const formRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (formRef.current) {
+      formRef.current.focus();
+    }
+  }, [todos]);
+
+  const isAllTodoCompleted = todos.every(todo => todo.completed);
+
+  return (
+    <header className="todoapp__header">
+      <button
+        aria-label="toggle-all"
+        type="button"
+        className={cn('todoapp__toggle-all', {
+          active: isAllTodoCompleted,
+        })}
+        onClick={() => (
+          handleUpdate(
+            filterTodosByCompletion(todos)
+              .map(todo => todo.id),
+          ))}
+      />
```

Comment:
Let's create a separate handler function. 
I think would be better to create this function in the `app` and provide it via props